src/TestableIO.System.IO.Abstractions.TestingHelpers/MockDirectory.cs

```diff
@@ -110,7 +110,17 @@ public override IFileSystemInfo CreateSymbolicLink(string path, string pathToTar
         /// <inheritdoc />
         public override IDirectoryInfo CreateTempSubdirectory(string prefix = null)
         {
-            throw CommonExceptions.NotImplemented();
+            prefix ??= "";
+            string potentialTempDirectory;
+
+            // Perform directory name generation is a loop, just in case the randomly generated name already exists.
+            do
+            {
+                var randomDir = $"{prefix}{Path.GetRandomFileName()}";
+                potentialTempDirectory = Path.Combine(Path.GetTempPath(), randomDir);
+            } while (Exists(potentialTempDirectory));
+
+            return CreateDirectoryInternal(Path.Combine(Path.GetTempPath(), potentialTempDirectory));
```

Comment:
`potentialTempDirectory` already contains the `Path.GetTempPath()`, so you should not prefix it with the path to the temporary directory twice...

Reply:
Good catch. Fixed.